### ISSUE TITLE
[BUG] - Idempotence check fails in vultr.cloud.firewall_rule when ICMP has port

### ISSUE THREAD

**Issue by goetzk** [vultr/ansible-collection-vultr#76]
**Describe the bug**
When an ICMP firewall rule is created with a port on subsequent runs the API returns an error.

This behaviour holds true if the port is -1 or 0.

**To Reproduce**

Playbook:
```
- name: Test firewall behaviour
  hosts: localhost
  gather_facts: false
  vars:
    instance_hostname: test.collectiveaccess.au
    default_firewall_rule_entries:
      - subnet: "0.0.0.0"
        port_min: "80"
        port_max: "1024"
        subnet_size: "0"
        ip_gen: v4
      - subnet: "0.0.0.0"
        port_min: "443"
        port_max: "1024"
        subnet_size: "0"
        ip_gen: v4
      - subnet: '0.0.0.0'
        port_min: '-1'
        subnet_size: "0"
        proto: 'icmp'
        ip_gen: v4
  tasks:
    - name: Ensure firewall rule group is configured for this host
      vultr.cloud.firewall_group:
        api_key: "{{ vultr_api_key }}"
        description: "{{ 'Group to hold all rules associated with ' + instance_hostname }}"
        state: present
      register: firewall_group_result
    - name: Ensure firewall rules exist
      vultr.cloud.firewall_rule:
        api_key: "{{ vultr_api_key }}"
        group: "{{ firewall_group_result.vultr_firewall_group.description }}"
        ip_type: "{{ item.ip_gen }}"
        notes: "Access to port range {{ item.port_min }} through {{ item.port_max |default(item.port_min) }} from {{ item.subnet }}/{{ item.subnet_size }}"
        port: "{{ item.port_min + ':' + item.port_max |default(item.port_min) }}"
        protocol: "{{ item.proto |default('tcp') }}"
        state: present
        subnet: "{{ item.subnet }}"
        subnet_size: "{{ item.subnet_size }}"
      when: firewall_group_result.vultr_firewall_group.description is defined
      loop: "{{ default_firewall_rule_entries }}"
      register: firewall_rules_result
      # Ignoring errors until I can come back to this and fix it properly
      ignore_errors: true
```

Result of `ansible-playbook test-firewall.yaml `:
```
PLAY [Test firewall behaviour] *************************************************************************************************************************************************************************************************************

TASK [Ensure firewall rule group is configured for this host] ******************************************************************************************************************************************************************************
ok: [localhost]

TASK [Ensure firewall rules exist] *********************************************************************************************************************************************************************************************************
ok: [localhost] => (item={'subnet': '0.0.0.0', 'port_min': '80', 'port_max': '1024', 'subnet_size': '0', 'ip_gen': 'v4'})
ok: [localhost] => (item={'subnet': '0.0.0.0', 'port_min': '443', 'port_max': '1024', 'subnet_size': '0', 'ip_gen': 'v4'})
failed: [localhost] (item={'subnet': '0.0.0.0', 'port_min': '-1', 'subnet_size': '0', 'proto': 'icmp', 'ip_gen': 'v4'}) => {"ansible_loop_var": "item", "changed": false, "fetch_url_info": {"body": "{\"error\":\"This rule is already defined\",\"status\":400}", "connection": "close", "content-type": "application/json", "date": "Tue, 20 Jun 2023 03:32:20 GMT", "msg": "HTTP Error 400: Bad Request", "server": "nginx", "status": 400, "strict-transport-security": "max-age=63072000", "transfer-encoding": "chunked", "url": "https://api.vultr.com/v2/firewalls/[removed]/rules", "x-content-type-options": "nosniff", "x-frame-options": "DENY", "x-robots-tag": "noindex,noarchive", "x-user": "[removed]"}, "item": {"ip_gen": "v4", "port_min": "-1", "proto": "icmp", "subnet": "0.0.0.0", "subnet_size": "0"}, "msg": "Failure while calling the Vultr API v2 with POST for \"/firewalls/[removed]/rules\"."}
...ignoring

PLAY RECAP *********************************************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=1   
```

If port is commented out in `vultr.cloud.firewall_rule` all rules check OK:
```
PLAY [Test firewall behaviour] *************************************************************************************************************************************************************************************************************

TASK [Ensure firewall rule group is configured for this host] ******************************************************************************************************************************************************************************
ok: [localhost]

TASK [Ensure firewall rules exist] *********************************************************************************************************************************************************************************************************
ok: [localhost] => (item={'subnet': '0.0.0.0', 'port_min': '80', 'port_max': '1024', 'subnet_size': '0', 'ip_gen': 'v4'})
ok: [localhost] => (item={'subnet': '0.0.0.0', 'port_min': '443', 'port_max': '1024', 'subnet_size': '0', 'ip_gen': 'v4'})
ok: [localhost] => (item={'subnet': '0.0.0.0', 'port_min': '-1', 'subnet_size': '0', 'proto': 'icmp', 'ip_gen': 'v4'})

PLAY RECAP *********************************************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0 
```

**Expected behavior**
Port ignored and ICMP check passed

**Desktop (please complete the following information where applicable:**
 - OS: Debian oldstable ansible controller.
 - Latest release of ansible.collection.vultr installed.

**Comment by goetzk** [comment]
It occurred to me to try and and run without code trying to make the port a range but just hard coding a single port with ` port: "{{ item.port_min }}"`.
That still results in the failure seen above.

**Comment by resmo** [comment]
So port has no function for anything else then tcp/udp, so I am considering to either show an error "port must not be set for anyhting esle then tcp/udp" or a warning, that port is ignored. 

In case we show an error, your playbook would still not run, so I would change it to:

```yaml
port: "{{ item.port_min + ':' + item.port_max | default(item.port_min) if item.port_min is defined else omit }}"

```

This should omit the `port` param if no var `port_min` is set.

**Comment by resmo** [comment]
Preparing fix in #77
